scrapy/utils/retry.py

```diff
@@ -0,0 +1,54 @@
+from scrapy.utils.python import global_object_name
+
+
+class RetryHandler(object):
+    """utilities to handle retries"""
+
+    def __init__(self, spider, request):
+        self.spider = spider
+        self.request = request
+        self.max_retry_times = spider.settings.getint('RETRY_TIMES')
```

Comment:
Maybe setting default values, or raising an specific exception if these settings aren't available.

Reply:
It should be possible to override `max_retry_times` per request.